cabal-testsuite/PackageTests/RtsOptsClean/setup.test.hs

```diff
@@ -0,0 +1,6 @@
+import Test.Cabal.Prelude
+
+-- Test that setup shows all the 'autogen-modules' warnings.
+main = setupAndCabalTest $ do
+    setup' "configure" [] >>=
+         assertOutputDoesNotContain "Warning: Instead of 'ghc-options: -I0' use 'include-dirs: 0'"
```

Comment:
That's the behaviour without your fix?

Reply:
Yes. This is documented in #4255 and reproduces today.